backend/src/main/java/com/github/johannesthorbergsson/backend/workshops/WorkshopService.java

```diff
@@ -18,28 +19,42 @@ public class WorkshopService {
     public List<Workshop> getAllWorkshops() {
         return workshopRepository.findAll();
     }
+    public Workshop getWorkshopById(String id){
+        return workshopRepository.findById(id).orElseThrow(NoSuchWorkshopException::new);
+    }
+
+    public List<Workshop> workshopSearch(String searchTerm){
+        String optimizedSearchTerm = searchTerm.toLowerCase().trim();
+        return workshopRepository.findAll().stream().filter(workshop ->
+               workshop.location().toLowerCase().trim().contains(optimizedSearchTerm)
+            || workshop.services().stream().anyMatch(service -> service.toLowerCase().trim().contains(optimizedSearchTerm))
+            || workshop.inventory().stream().anyMatch(component ->
+                       component.category().toLowerCase().trim().contains(optimizedSearchTerm))
+            || workshop.inventory().stream().anyMatch(component ->
+                       component.type().toLowerCase().trim().contains(optimizedSearchTerm))
+            || workshop.name().toLowerCase().trim().contains(optimizedSearchTerm)).toList();
+    }
     public Workshop addWorkshop(Principal principal, WorkshopRequest workshopRequest){
         if(!userService.getCurrentUser(principal).role().equals("WORKSHOP")) {
            throw new UnauthorizedAccessException();
         }
         Workshop newWorkshop = new Workshop(
                 workshopRequest.id(),
                 workshopRequest.name(),
-                principal.getName(),
                 workshopRequest.location(),
                 workshopRequest.coordinates(),
                 workshopRequest.services(),
                 workshopRequest.inventory());
         return workshopRepository.save(newWorkshop);
     }
     public WorkshopResponse updateWorkshop(String id, WorkshopRequest workshopRequest, Principal principal){
-        if(!workshopRepository.findById(id).orElseThrow(NoSuchWorkshopException::new).username().equals(principal.getName())){
+        UserResponse user = userService.getCurrentUser(principal);
+        if(!workshopRepository.findById(id).orElseThrow(NoSuchWorkshopException::new).id().equals(user.id())){
```

Comment:
I recommend to split this into two lines so it is a lot easier to read :)
```suggestion
        Workshop workshop = getWorkshopById(id);
        if(!workshop.id().equals(user.id())){
```

Reply:
Actually now that I re-read it... You don't really care about the `workshop` object, do you? You only want to raise the exception if needed. So you could also do:
```suggestion
        getWorkshopById(id);
        if(!id.equals(user.id())){
```